src/core/dice.c

```diff
@@ -302,6 +303,93 @@ enclave_attester_err_t dice_generate_evidence_buffer_with_tag(
 	return ret;
 }
 
+enclave_attester_err_t dice_generate_endorsements_buffer_with_tag(
+	const char *type, const attestation_endorsement_t *endorsements,
+	uint8_t **endorsements_buffer_out, size_t *endorsements_buffer_size_out)
+{
+	enclave_attester_err_t ret;
+	cbor_item_t *root = NULL;
+	cbor_item_t *array = NULL;
+
+	/* Currently we only support endorsements for SGX/TDX ECDSA mode */
+	if (strcmp(type, "sgx_ecdsa") && strcmp(type, "tdx_ecdsa")) {
```

Comment:
Since we will add csv,sev,snp etc tee type in this function. I think use the following code should be better.
```sh
if (!strcmp(type, "sgx_ecdsa") || !strcmp(type, "tdx_ecdsa")) {
// implemtation
}
else if (!strcmp(type, "csv")) {
// TODO
}

```